### PR TITLE
MGMT-1128 Every execute command that will fail - will change the host…

### PR DESCRIPTION
**Pull request by gamli75**
… status to error

@ronniel1 @eranco74 @filanov @ori-amizur please review.
I added a subsystem test to verify it.

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1045,11 +1045,11 @@ func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer
 	return installer.NewPostStepReplyNoContent()
 }
 
-func handleReplyError(params installer.PostStepReplyParams, b *bareMetalInventory, ctx context.Context, host *models.Host) error {
+func handleReplyError(params installer.PostStepReplyParams, b *bareMetalInventory, ctx context.Context, h *models.Host) error {
 
-	if strings.HasPrefix(params.Reply.StepID, string(models.StepTypeExecute)) {
+	if strings.HasPrefix(params.Reply.StepID, host.InstallCmdStep) && strings.Contains(params.Reply.StepID, string(models.StepTypeExecute)) {
```
Comment (ronniel1) [inline internal/bminventory/inventory.go:1050]: We need to know that this is an "install" command.
No need to check the StepTypeExecute string. It's redundant

File: internal/host/installcmd.go [hunk 2]
```diff
@@ -34,6 +35,7 @@ func (i *installCmd) GetStep(ctx context.Context, host *models.Host) (*models.St
 	step := &models.Step{}
 	step.StepType = models.StepTypeExecute
 	step.Command = "bash"
+	step.StepID = fmt.Sprintf("%s-%s-%s", InstallCmdStep, models.StepTypeExecute, uuid.New().String()[:8])
```
Comment (ronniel1) [inline internal/host/installcmd.go:38]: no need to the execute
Install will alway use execute

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (gamli75) [reply internal/bminventory/inventory.go:1050]: fixed

File: internal/host/installcmd.go (hunk 2 above)
Comment (gamli75) [reply internal/host/installcmd.go:38]: @ori-amizur are you using the step type in the step ID somewhere in your code? can I remove the stepType execute from the step ID?

Comment (ori-amizur) [reply internal/host/installcmd.go:39]: I just take the step id from get next steps reply and move it the step reply without modification.
So you can remove the execute from step id.
I just rely on the step type.

**Pull request merged**